File: BackEnd/Infrastructure/Persistance/EntityFramework/Broker.cs
namespace MarketDataSync.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("Broker")]
    public partial class Broker
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Broker()
        {
            BrokerConfigurations = new HashSet<BrokerConfiguration>();
        }

        public Guid BrokerId { get; set; }

        [Required]
        [StringLength(50)]
        public string Name { get; set; }

        [StringLength(250)]
        public string Description { get; set; }

        [StringLength(250)]
        public string APIDocumentationUrl { get; set; }

        [StringLength(250)]
        public string AuthenticationUrl { get; set; }

        public bool IsActive { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<BrokerConfiguration> BrokerConfigurations { get; set; }
    }
}

File: FrontEnd/src/app/broker/model/broker.ts
export class Broker {
    public brokerId?: string;
    public name!: string;
    public description!: string;
    public apiDocumentationUrl!: string;
    public authenticationUrl!: string;
    public isActive!: boolean;
    public brokerConfigurations: BrokerConfiguration[];
  
    constructor() {
      this.brokerConfigurations = [];
    }
  }
  
  export class BrokerConfiguration {
    public brokerConfigurationId?: string;
    public brokerId?: string;
    public configKey!: string;
    public needsToUpdateDaily: boolean;
  
    constructor() {
      this.needsToUpdateDaily = false;
    }
  }